components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandler.java

```diff
@@ -296,8 +302,8 @@ public OAuth2AuthorizeRespDTO issue(OAuthAuthzReqMessageContext oauthAuthzMsgCtx
             // Default token validity Period
             long validityPeriodInMillis = OAuthServerConfiguration.getInstance().
                     getUserAccessTokenValidityPeriodInSeconds() * 1000;
-            if (spTimeConfigObj.getUserAccessTokenExpiryTime() != null) {
-                validityPeriodInMillis = spTimeConfigObj.getUserAccessTokenExpiryTime();
+            if (oAuthAppDO.getUserAccessTokenExpiryTime() != 0) {
```

Comment:
Do we allow to add validity period as 0 ? If so what does it mean?

Reply:
This meant that, if the user needs no expiry time(infinite time) we have the -1 configuration in file. but here if the user entered 0 since it is meaningless we read default expiry times from file